heron/scheduler-core/tests/java/com/twitter/heron/scheduler/SubmitterMainTest.java

```diff
@@ -181,8 +183,31 @@ public void testSubmitTopologyLauncherException() throws Exception {
     submitterMain.submitTopology();
   }
 
+  @Test(expected = SubmitDryRunResponse.class)
+  public void testSubmitTopologyDryRun() throws Exception {
+    SchedulerStateManagerAdaptor adaptor = mock(SchedulerStateManagerAdaptor.class);
+    PowerMockito.whenNew(SchedulerStateManagerAdaptor.class).withAnyArguments().
+        thenReturn(adaptor);
+    SubmitterMain submitterMain = spy(new SubmitterMain(config, topology));
+    PowerMockito.mockStatic(Context.class);
+    PowerMockito.when(Context.dryRun(config)).thenReturn(true);
+    doThrow(new SubmitDryRunResponse(topology, config, mock(PackingPlan.class))).
```

Comment:
aren't you trying to assert that the code throws this response without mocking it's behavior to do so?